### DISCUSSION TITLE
Icicle Kit latest design has a AXI Address SHIM added to it

### DISCUSSION THREAD

**Discussion by josh25219** [polarfire-soc/polarfire-soc-documentation#39] [Reference designs, FPGA and development kits] [answered]
1) What the reason is for the AXI Address SHIM between the PCIE master interface and FIC0 target interface?
2) The AXI interconnect between the PCIE Master interface and the FIC0 target has only 1 master and 1 slave.   Could this interface be directly connected without the AXI interconnect?   Why is the AXI interconnect needed?

Thanks,
Josh

**Comment by hughbreslin** [comment, marked as answer]
Hi @josh25219 

For number 1 theres a few parts to it (this [document](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/fundamentals/pcie/mpfs-memory-configuration.md) is quite useful):

Our PCIe block only has 32 address bits available to the FPGA fabric, but we're primarily using DDR with 38 address bits in Linux - this means we have an issue addressing the memory used by Linux when doing transfers using PCIe. To resolve this we use address translation in the PCIe block to subtract the 38 bit DDR base address from the src / destination address of the PCIe transfer so it can get out of the 32 address bit interface and then use the SHIM to add back on the 38bit DDR base address in the fabric. Basically the addresses coming out of PCIe are just offsets to locations in DDR memory. For example:

Linux tells an endpoint to transfer data to / from 0x10_0001_F000
Address translation in the PCIe block subtracts 0x10_0000_0000 from this address making it 0x0001_F000
The address passes through the address SHIM and becomes 0x10_0001_F000 again

For number 2, yes this is technically possible but I think it would be very difficult to close timing in this configuration. FIC1 is physically closer to the PCIe block (and we will be transitioning to use FIC1 for PCIe transactions overall) but again I'm not sure if you'd be able to get timing closure.

**Reply by josh25219** [reply]
Thanks Hugh!
